v3/src/components/case-tile-common/inspector-panel/case-tile-inspector.tsx

```diff
@@ -36,15 +39,55 @@ export const CaseTileInspector = ({ tile, show, showResizeColumnsButton }: IProp
       case "datasetInfo":
         setShowInfoModal(true)
         break
-      case "resizeColumns":
-        //TODO move log to respective handler
-        caseTileModel?.applyModelChange(() => {}, {
-          log: logStringifiedObjectMessage("resizeColumns: %@", {dataContext: data?.name}, "table")
-        })
-        break
     }
   }
 
+  const resizeAllColumns = () => {
+    data?.collections.forEach((collection) => {
+      collection.attributes.forEach((attr) => {
+        if (attr) {
+          const attrId = attr?.id
+          const longestContentWidth = findLongestContentWidth(attr)
+          tableModel?.applyModelChange(() => {
+            tableModel?.setColumnWidth(attrId, longestContentWidth)
+          }, {
+            log: {message: "Resize all columns", args:{}, category: "table"},
+            undoStringKey: "DG.Undo.caseTable.resizeColumns",
+            redoStringKey: "DG.Redo.caseTable.resizeColumns"
+          })
```

Comment:
You should wrap the entire loop in `applyModelChange`, not each column. As is, you'll send a log message and create an undoable action for each attribute in the dataset.